apps/ui-builder/src/components/ComponentTreeItem.tsx

```diff
@@ -0,0 +1,220 @@
+  handleDeleteSelected,
+  handleSelectComponent,
+  handleClone,
+  handleMove,
+}) => {
+  const styles = useStyles();
+
+  const [deletePopoverOpened, setDeletePopoverOpened] = React.useState(false);
+  const [onHover, setOnHover] = React.useState(false);
+
+  const openPopover = React.useCallback(() => setDeletePopoverOpened(true), [setDeletePopoverOpened]);
+  const closePopover = React.useCallback(() => setDeletePopoverOpened(false), [setDeletePopoverOpened]);
+
+  const selectItem = React.useCallback(() => {
+    if (selected) {
+      return;
+    }
+    handleSelectComponent(node.element);
+  }, [handleSelectComponent, node, selected]);
+
+  const onKeyDown = React.useCallback(
+    e => {
+      // enter or space
+      if (e.key === 'Enter' || e.keyCode === 32) {
+        selectItem();
+        return;
+      }
+
+      if (e.key === 'ArrowDown') {
+        e.target.nextSibling && e.target.nextSibling.focus();
+        return;
+      }
+      if (e.key === 'ArrowUp') {
+        e.target.previousSibling && e.target.previousSibling.focus();
+        return;
+      }
+
+      if (isMac) {
+        macKeyDown(e, e.target);
+      }
+    },
+    [selectItem],
+  );
+
+  React.useEffect(() => {
+    if (!selected) {
+      setDeletePopoverOpened(false);
+    }
+  }, [selected]);
+
+  const deleteButton = React.useMemo(
+    () => (
+      <Popover open={deletePopoverOpened}>
+        <PopoverTrigger>
+          <Tooltip relationship="label" content="Delete" withArrow>
+            <Button
+              icon={<DeleteRegular />}
+              appearance="subtle"
+              size="small"
+              onClick={openPopover}
+              className={styles.button}
+            />
+          </Tooltip>
+        </PopoverTrigger>
+        <PopoverSurface className={styles.deletePopover}>
+          <span className={styles.deletePopoverCaption}>Are you sure you want to delete {node.title}?</span>
+          <Button onClick={closePopover}>Cancel</Button>
+          <Button appearance="primary" onClick={handleDeleteSelected}>
+            Delete
+          </Button>
+        </PopoverSurface>
+      </Popover>
+    ),
+    [node, deletePopoverOpened, openPopover, closePopover, handleDeleteSelected, styles],
+  );
+
+  const onMouseEnter = React.useCallback(() => setOnHover(true), [setOnHover]);
+  const onMouseLeave = React.useCallback(() => setOnHover(true), [setOnHover]);
+
+  const addAfter = React.useCallback(() => handleAddComponent(node.id, 'after'), [handleAddComponent, node.id]);
+  const addBefore = React.useCallback(() => handleAddComponent(node.id, 'before'), [handleAddComponent, node.id]);
+  const addChild = React.useCallback(() => handleAddComponent(node.id, 'child'), [handleAddComponent, node.id]);
+  const remove = React.useCallback(() => handleDeleteComponent(node.id), [handleDeleteComponent, node.id]);
+
+  return (
+    <Menu openOnContext={true}>
+      <MenuTrigger>
+        <div
+          tabIndex={0}
```

Comment:
making divs focusable is tricky. You did great job at adding keyboarding, however you also need:
- add appropriate role for screen reader users
- add focus indicator, as not all browsers yet fully support focus indicators with sufficient contrast by default

Reply:
Removed focusability in favor of tree behavior